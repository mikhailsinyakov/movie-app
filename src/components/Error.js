import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withLanguageContext } from "./Language";

const Error = ({ getUIText, className }) => (
  <div className={className}>
    <img src="img/error.png" alt="error" />
    <h2>{getUIText("errorTitle")}</h2>
    <p>{getUIText("errorProblem")}</p>
    <p>{getUIText("errorSolution")}</p>
  </div>
);

Error.propTypes = {
  getUIText: PropTypes.func.isRequired
};

const StyledError = styled(Error)`
  background-color: rgba(241, 209, 209, 0.8);
  padding: 1rem 1.5rem;
  border-radius: 1rem;
  margin: 1rem 0.5rem;
  text-align: center;
  border: 0.5rem solid rgba(189, 148, 148, 0.8);
  max-width: 300px;

  @media screen and (min-width: 400px) {
    & {
      margin: 1rem auto;
    }
  }
`;

export default withLanguageContext(StyledError);
