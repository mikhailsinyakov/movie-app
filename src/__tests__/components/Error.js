import React from "react";
import Renderer from "react-test-renderer";
import { render, getByAltText } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import withLanguage from "components/Language";
import Error from "components/Error";

const Parent = ({ children }) => <div>{children}</div>;
const ParentWithLanguage = withLanguage(Parent);

describe("Snapshot", () => {
  let component;
  beforeEach(() => {
    component = Renderer.create(
      <ParentWithLanguage>
        <Error />
      </ParentWithLanguage>
    );
  });
  afterEach(() => {
    component.unmount();
  });
  it("matches snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const renderComponent = () =>
  render(
    <ParentWithLanguage>
      <Error />
    </ParentWithLanguage>
  );

describe("Error", () => {
  it("renders error message and error image", () => {
    const { getByText, getByAltText } = renderComponent();
    expect(getByText(/oops/i)).toBeInTheDocument();
    expect(getByText(/something went wrong/i)).toBeInTheDocument();
    expect(getByText(/failed to fetch data/i)).toBeInTheDocument();
    expect(getByText(/try to come to the site later/i)).toBeInTheDocument();
    expect(getByAltText(/error/i)).toBeInTheDocument();
  });
});
