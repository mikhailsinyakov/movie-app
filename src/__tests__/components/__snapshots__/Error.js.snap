// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches snapshot 1`] = `
<div>
  <div
    className="sc-AxjAm itFQbf"
  >
    <img
      alt="error"
      src="img/error.png"
    />
    <h2>
      Oops! Something went wrong!
    </h2>
    <p>
      We encountered an error while trying to fetch the data from third-party resource.
    </p>
    <p>
      Try go to the website later
    </p>
  </div>
</div>
`;
